Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,199 @@
+                }
+                else if (args.State.IsDead())
+                {
+                    var guard = EntityManager.GetEntity(component._hostedguardian);
+                    SoundSystem.Play(Filter.Pvs(guard), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.25f));
+                    EntityManager.QueueDeleteEntity(guard);
+                    EntityManager.QueueDeleteEntity(uid);
+                }
+            } 
+        }
+
+        private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, DamageChangedEvent args)
+        {
+            var guardiandamage = EntityManager.GetEntity(uid).EnsureComponent<DamageableComponent>();
+            var hostdamage = EntityManager.GetEntity(component.Host).EnsureComponent<DamageableComponent>();
+            if (args.DamageDelta != null)
+            {
+                EntitySystem.Get<DamageableSystem>().SetDamage(hostdamage, (hostdamage.Damage + args.DamageDelta * component.DamagePercent));
+                hostdamage.Owner.PopupMessage(Loc.GetString("guardian-entity-taking-damage"));
+            }
+        }
+
+        /// <summary>
+        /// Triggers while trying to examine an activator to see if it's used
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnActivatorExamined(EntityUid uid, GuardianCreatorComponent component, ExaminedEvent args)
+        {
+           if (component.Used == true)
+           {
+               string usedstring = Loc.GetString("guardian-activator-empty-invalid-creaton");
+               args.Message.AddMarkup("\n" + $"[color=#ba1919]{usedstring}[/color]");
+           }
+        }
+
+        /// <summary>
+        /// Adds the guardian host component to the user and spawns the guardian inside said component
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnGuardianCreated(EntityUid uid, GuardianCreatorComponent component, UseInHandEvent args)
+        {
+            //Only works if the guardian isn't already present. NOTE: this is up to preference, we can probably later allow
+            //a person to host multiple guardians, however the idea is that in a normal setting you won't get the opportunity to
+            if (!args.User.TryGetComponent<GuardianHostComponent>(out var guardianComponent))
+            {
+                if (args.User.TryGetComponent<SharedActionsComponent>(out SharedActionsComponent? action))
+                {
+                    if (component.Used == false)
+                    {
+                        var hostcomp = args.User.AddComponent<GuardianHostComponent>();
+                        var guardian = EntityManager.SpawnEntity(component.GuardianType, hostcomp.Owner.Transform.Coordinates);
```

Comment:
Would be ideal to have a test to validate prototypes because this has the chance of causing an exception at runtime.

Reply:
How?

Reply:
Just use an ensurecomponent and not add for the component for now.

Reply:
Woops replied to the wrong one; AFAIK we still don't have the EntityPrototype support for datafield so just add a TryIndex<EntityPrototype>(component.GuardianType) above it.